pkg/client/introduce/client.go

```diff
@@ -53,176 +38,63 @@ func New(ctx Provider, inv *didexchange.Invitation) (*Client, error) {
 		return nil, errors.New("cast service to Introduce Service failed")
 	}
 
-	store, err := ctx.StorageProvider().OpenStore(introduce.Introduce)
-	if err != nil {
-		return nil, err
-	}
-
 	return &Client{
-		Event:      introduceSvc,
-		service:    introduceSvc,
-		store:      store,
-		defaultInv: inv,
-		newUUID:    func() string { return uuid.New().String() },
+		Event:   introduceSvc,
+		service: introduceSvc,
 	}, nil
 }
 
-// InvitationEnvelope is a helper function that returns the dependency needed for the
-// service to proceed with the protocol. Dependency should be passed to the service through `Continue` function.
-// The function should never return an error. Instead of an error we provide a callable interface
-// and a state machine that will act according to the provided data.
-// Dependency is populated after executing the following functions:
-//  - SendProposal
-//  - SendProposalWithInvitation
-//  - HandleRequest
-//  - HandleRequestWithInvitation
-// usage: e.Continue(c.InvitationEnvelope(threadID))
-func (c *Client) InvitationEnvelope(thID string) *InvitationEnvelope {
-	opts, err := c.getInvitationEnvelope(thID)
-
-	if errors.Is(err, storage.ErrDataNotFound) {
-		return &InvitationEnvelope{Inv: c.defaultInv}
-	}
+// SendProposal sends a proposal to the introducees (the client does not have a public Invitation).
+func (c *Client) SendProposal(recipient1, recipient2 *introduce.Recipient) error {
```

Comment:
The introduce service dependency(`introduce.Recipient`) shouldn't be exposed to the users of this client. We can wrap the service models in client package similar to [DIDExchange](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/client/didexchange/models.go#L52).